src/lib/journey/journey.store.ts

```diff
@@ -3,13 +3,61 @@ import type { StepOptions } from '@forgerock/javascript-sdk/lib/auth/interfaces'
 import { writable, type Writable } from 'svelte/store';
 
 import { htmlDecode } from '$journey/_utilities/decode.utilities';
-import type { JourneyStore, JourneyStoreValue, StepTypes } from './journey.interfaces';
+import type { JourneyStore, JourneyStoreValue, StackStore, StepTypes } from './journey.interfaces';
 import { interpolate } from '$lib/_utilities/i18n.utilities';
 import {
   authIdTimeoutErrorCode,
   shouldPopulateWithPreviousCallbacks,
 } from './_utilities/step.utilities';
 
+function initializeStack(initOptions?: StepOptions) {
+  const initialValue = initOptions ? [initOptions] : [];
+  const { update, set, subscribe }: Writable<StepOptions[]> = writable(initialValue);
+
+  // Assign to exported variable (see bottom of file)
+  stack = {
+    pop: async (): Promise<StepOptions[]> => {
+      return new Promise((resolve) => {
+        update((current) => {
+          let state;
+          if (current.length) {
+            state = current.slice(0, -1);
+          } else {
+            state = current;
+          }
+          resolve([...state]);
+          return state;
+        });
+      });
+    },
+    push: async (options?: StepOptions): Promise<StepOptions[]> => {
```

Comment:
No Svelte specific anything in here, really, other than the core store APIs. Are you just wanting to remove the unnecessary braces?

Reply:
Oh, you're also wanting to remove the `new Promise`? I need to look closer at the `update` function that comes from the Svelte store. I think it's asynchronous, which is why I'm using the Promise. But, I'll see if I can make sure.

Reply:
it just seems redundant since its async already?

Reply:
Okay, that's right! The `update` function doesn't return anything, regardless of it being sync or async. Here's the `update` method's type:

```
const update: (this: void, updater: Updater<StepOptions[]>) => void
```

All we can do is return the value to the callback function _within_ the `update`, but not return a value from `update` itself. If that makes any sense.

Reply:
BTW, by async, I don't mean `async-await`, I just mean from the generic since of asynchronous code as it uses the callback pattern and update doesn't have an actual return value we can control.